clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java

```diff
@@ -202,7 +205,23 @@ public String toString() {
          * Mark that we want data from this partition in the upcoming fetch.
          */
         public void add(TopicPartition topicPartition, PartitionData data) {
+            if (sessionPartitions.containsKey(topicPartition)) {
+                altered.add(topicPartition);
+            } else {
+                added.add(topicPartition);
+            }
+
             next.put(topicPartition, data);
+            sessionPartitions.put(topicPartition, data);
+        }
+
+        /**
+         * Mark that we want to exclude this partition in the upcoming fetch.
+         */
+        public void remove(TopicPartition topicPartition) {
+            removed.add(topicPartition);
+            next.remove(topicPartition);
```

Comment:
Will any of these partitions be in next? My presumption is that they would not be in partitionStates and would not be in next if they are being removed. I think they would be in sessionPartitions though.